### ISSUE TITLE
MaxConstraint assertion violation

### ISSUE THREAD

**Issue by kjulian3** [NeuralNetworkVerification/Marabou#291]
I've noticed a strange issue that sometimes happens when using Maraboupy to evaluate a random point of a network that uses with max-pool. Sometimes, Marabou will reach an assertion violation on line 275 of MaxConstraint.cpp. The error seems to occur during pre-processing, but it rarely errors. In the script below, I'm generating the same input query and solving it 500 times, and usually at some point before reaching 500 an assertion violation occurs.

The error occurs in both Release and Debug modes. I've done a little debugging, and found that the resetMaxConstraint() function is getting called when _assignments is empty while _elements has two or three entries. When Marabou does not error, resetMaxConstraint is never called; only when maxErased is true and resetMaxConstraint is called does Marabou violate the assertion. I'm not sure how to debug the issue further and would appreciate some help.

This issue has sometimes caused Travis's python tests to fail, though rerunning the tests usually succeeds since the failure happens rarely.

Example python code to reproduce the issue:
```
from maraboupy import Marabou
import numpy as np
import os

# Global settings
OPT = Marabou.createOptions(verbosity = 2)  # Turn off printing
FG_FOLDER = "resources/tf/frozen_graph/"    # Folder for test networks written in frozen graph format
RANDOM_SEED = 1234                          # Seed random numbers for repeatability

"""
Test a convolutional network using max pool
"""
filename = os.path.join(os.path.dirname(__file__), FG_FOLDER, "conv_mp2.pb")
network = Marabou.read_tf(filename)
np.random.seed(RANDOM_SEED)
testInput = [np.random.random(inVars.shape) for inVars in network.inputVars]
for i in range(500):
    network.evaluateWithMarabou(testInput, options = OPT, filename = "")
    print(i)
```

**Comment by kjulian3** [comment]
I've also been able to reproduce this issue with an onnx network by setting 
```
filename = "resources/onnx/conv_mp1.onnx"
network = Marabou.read_onnx(filename)
```
in place of the tensorflow filename and read_tf functions. The onnx network is smaller, with fewer variables and equations, so it may be a better test network.

I've tried saving the input query using MarabouCore.saveQuery(), but the resulting file from a query that caused an error is identical to one that was produced by a query that didn't cause an error. I have also been unable to reproduce the error when using MarabouCore.loadQuery() to load the query file and solve with Marabou, even when it was written by a query that caused a failure.
